### ISSUE TITLE
使用train_embedding模块，微调bge-m3，只使用train_dataset不配置val_dataset一切正常，如果配置了，训练速度会慢几个数量级，甚至直接卡死

### ISSUE THREAD

**Issue by TestNLP** [NovaSearch-Team/RAG-Retrieval#111]
环境是：三张4090GPU,24G 显存。

save_steps: 10000
eval_steps: 100
log_interval: 5
neg_nums: 20
query_max_len: 64
passage_max_len: 64
shuffle: true

save_on_epoch_end: 1
num_max_checkpoints: 5

temperature: 0.02
epochs: 2
lr: 3e-5
batch_size: 32
seed: 666
warmup_proportion: 0.0001
gradient_accumulation_steps: 2
mixed_precision: bf16
all_gather: True
gradient_checkpointing: True

use_mrl: false
mrl_dims: "128, 256, 512, 768, 1024, 1280, 1536, 1792"

数据格式如下：
{"query": "p一二证书包含私钥跟公钥", "pos": ["不含私钥的证书"], "scores": [0.3456]}
{"query": "把数学调整为王广宇老师的班", "pos": ["将我的数学s课程调整为周四房亮老师课程", "麻烦后面帮忙调整周三王宇老师的课", "把我的数学调整到宁大侠 周四晚上的课", "老师，请帮我把数学S班调整为S+班", "数学目前的春上春下A+班的课程调整到周三的王宝金老师S班，班主任罗怀琦", "老师您好，我想把数学调整一下难度，您帮我换到高三数学A+班，王子悦老师的课", "您好，请把我的数学春季课调换到     王子悦老师周六18:30-20:40的A+班。", "麻烦把春季数学课从博宇老师调到泽轩老师的周天10点半的课，辅导老师思源老师"], "scores": [0.84428, 0.847205, 0.848018, 0.866069, 0.849287, 0.852362, 0.86873, 0.813492]}
{"query": "激活码无法在该校激活", "pos": ["为什么显示激活码在该校无法激活", "学校给的激活码用不了", "有激活码为什么显示的是无法在本学校激活"], "scores": [0.9789, 0.9623, 0.9371]}

**Comment by NLPJCL** [comment]
数据中有scores的话，这是想蒸馏吗？
这种建议query对应的pos的数量要多一点并且pos数量得相同吧，不然会有问题。

**Comment by NLPJCL** [comment]
val_dataset，时间会变长，这个确实不太符合预期，你有排查出来原因吗？

**Comment by TestNLP** [comment]
pos数量确实不一样，少的一个，多的有100个，没有排查出原因。

**Comment by NLPJCL** [comment]
@Bool1020 有空可以扫一眼，我看代码没看出什么问题会导致这个现象。

**Comment by Bool1020** [comment]
你的验证集大小是不是太大了（条数太多了），每次验证会把整个数据集推理一遍 @TestNLP
